config/v1/types_ingress.go

```diff
@@ -152,8 +153,53 @@ type AWSIngressSpec struct {
 	// +kubebuilder:validation:Required
 	Type AWSLBType `json:"type,omitempty"`
+
+	// networkLoadBalancerParameters holds configuration parameters for an AWS
+	// Network Load Balancer. This field is present only if the type field is set to NLB.
+	//
+	// +openshift:enable:FeatureGate=SetEIPForNLBIngressController
+	// +optional
+	NetworkLoadBalancerParameters *AWSNetworkLoadBalancerParameters `json:"networkLoadBalancer,omitempty"`
 }
 
+// AWSNetworkLoadBalancerParameters holds configuration parameters for an
+// AWS Network load balancer. For Example: Setting AWS EIPs https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html
+type AWSNetworkLoadBalancerParameters struct {
+	// eipAllocations assign Elastic IP addresses to the Network Load Balancer.
+	// The number of Allocation IDs must match the number of subnets that are used for the load balancer.
+	// See https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html for general
+	// information about configuration, characteristics, and limitations of Elastic IP addresses.
+	// This field only specifies EIPs for the default IngressController. It is not a default value
+	// for all IngressControllers.
+	// +optional
+	// +listType=atomic
+	// +kubebuilder:validation:XValidation:rule=`self.all(x, self.exists_one(y, x == y))`,message="eipAllocations cannot contain duplicates"
+	// +kubebuilder:validation:MaxItems=10
+	EIPAllocations []EIPAllocation `json:"eipAllocations"`
+}
+
+// EIPAllocation is a reference to available eip allocations in AWS environment.
+// The pattern for an EIP allocation ID is:
+// - Starts with eipalloc-
+// - Followed by an alphanumeric string of exactly 17 characters
+// Explanation of the regex `^eipalloc-[0-9a-fA-F]{17}$` for validating value of the EIPAllocation:
+// - ^eipalloc- ensures the string starts with "eipalloc-".
+// - [0-9a-fA-F]{17} matches exactly 17 hexadecimal characters (0-9, a-f, A-F).
+// - $ ensures the string ends after the 17 hexadecimal characters.
+// Example of Valid and Invalid values:
+// - eipalloc-1234567890abcdef1 is valid.
+// - eipalloc-1234567890abcde is not valid (too short).
+// - eipalloc-1234567890abcdefg is not valid (contains a non-hex character 'g').
+// Max length is calculated as follows:
+// - eipalloc- = 9 chars and 17 hexadecimal chars after `-`
+// - So, total is 17 + 9 = 26 chars required for value of an EIPAllocation.
+// +kubebuilder:validation:MinLength=26
+// +kubebuilder:validation:MaxLength=26
+// +kubebuilder:validation:XValidation:rule=`!self.contains(',')`,message="eipAllocations cannot contain a comma"
+// +kubebuilder:validation:XValidation:rule=`self.startsWith('eipalloc-')`,message="eipAllocations should start with 'eipalloc-'"
+// +kubebuilder:validation:XValidation:rule=`self.split("-", 2)[1].matches('[0-9a-fA-F]{17}$')`,message="eipAllocations must be eipalloc- followed by exactly 17 hexadecimal characters (0-9, a-f, A-F)"
```

Comment:
I think this validation will reject `eipalloc-0123456789abcdef0-0123456` or `eipalloc-012345678-0123456`, but the reasons why are subtle as the rejection relies on the combination of validation rules.  I wonder whether it would be worth adding a couple more test cases?  
```yaml
    - name: Should not be able to create an ingress config in which the value of eipAllocations has a valid hexadecimal part followed by an extra hexadecimal part.
      initial: |
        apiVersion: config.openshift.io/v1
        kind: Ingress
        metadata:
          name: cluster
        spec:
          loadBalancer:
            platform:
              type: AWS
              aws:
                type: NLB
                networkLoadBalancer:
                  eipAllocations:
                  - eipalloc-0123456789abcdef0-0123456
      expectedError: "[spec.loadBalancer.platform.aws.networkLoadBalancer.eipAllocations[0]: Too long: may not be longer than 26"
    - name: Should not be able to create an ingress config in which the value of eipAllocations has a short hexadecimal part followed by an extra hexadecimal part.
      initial: |
        apiVersion: config.openshift.io/v1
        kind: Ingress
        metadata:
          name: cluster
        spec:
          loadBalancer:
            platform:
              type: AWS
              aws:
                type: NLB
                networkLoadBalancer:
                  eipAllocations:
                  - eipalloc-012345678-0123456
      expectedError: "eipAllocations must be eipalloc- followed by exactly 17 hexadecimal characters (0-9, a-f, A-F)"
```